bld/CLMBuildNamelist.pm

```diff
@@ -678,6 +678,11 @@ sub setup_cmdl_chk_res {
   }
 }
 
+sub begins_with
+{
```

Comment:
This is making a bunch of perl assumptions in regard to syntax that are unintuitive, so at least need some comments to explain.

Reply:
@slevis-lmwg see what I've added below to address this.

Reply:
I like it.